### PR TITLE
Read just part of a file

### PR DESCRIPTION
**Pull request by CharlesOkwuagwu**
Adds a convenience for reading just part of a file

http://stackoverflow.com/questions/37121605/how-to-read-a-portion-of-a-file-in-elixir-or-erlang?noredirect=1#comment61817174_37121605

### CODE REVIEW

**Comment by whatyouhide** [comment]
Hey @CharlesOkwuagwu, I think you pasted this code from some kind of diff and forgot to remove the `+` signs maybe? 😄

File: lib/elixir/lib/file.ex [hunk 1]
```diff
@@ -246,6 +246,18 @@ defmodule File do
     end
   end
 
+ @doc """
+ +  Returns `binary`, which is part of the contents of `path` starting at `start` and of length
+ +  `bytes`
+ +  """
+ +  @spec read(Path.t, pos_integer, pos_integer) :: binary 
+ +  def read(path, start, bytes) do
+ +    {ok, f} = :file.open(path, [:binary])
```
Comment (whatyouhide) [inline lib/elixir/lib/file.ex]: `ok` is not an Elixir atom, it's a variable. You probably meant `:ok` here (and below).

**Comment by CharlesOkwuagwu** [comment]
will fix now...

File: lib/elixir/lib/file.ex (hunk 1 above)
Comment (whatyouhide) [inline lib/elixir/lib/file.ex]: Also, maybe we can avoid using one-letter variables (`file` instead of `f`).

**Comment by whatyouhide** [comment]
Also, I just missed completely the fact that nobody discussed this feature, sorry! You may want to post feature discussions on the [elixir-lang-core](https://groups.google.com/forum/#!forum/elixir-lang-core) mailing list, where feature proposals should be posted so that the community can discuss them before a PR is submitted :)

**Comment by CharlesOkwuagwu** [comment]
@whatyouhide I submitted this to the core mailing list yesterday...

https://groups.google.com/forum/#!topic/elixir-lang-core/NfjQxv2hBUI

**Comment by whatyouhide** [comment]
@CharlesOkwuagwu yeah, updated my comment, sorry :)

**Comment by eksperimental** [comment]
You will also need to include tests.
I think it'd be better if you just send a link to the mailing with with the implementation linking to your fork.
that way we avoid duplicate discussions in the mailing list and here.

**Comment by josevalim** [comment]
Thanks for the PR @CharlesOkwuagwu!

As others have mentioned, such must be discussed first in the elixir-lang-core mailing list. If nobody replied yet, it is likely because there isn't much interest on this feature, which probably means it should not be added as part of the language (specially since it is already possible to implement it today).

**Comment by josevalim** [comment]
I am closing this for now. If others are involved and believe there is a consensus around this feature, then we can gladly revisit it.

**Comment by CharlesOkwuagwu** [comment]
Understood.

**Pull request closed without merging**